### PR TITLE
Fix compiler warnings

### PR DESCRIPTION
**Pull request by tamird**
Best to review commit-by-commit, since they each fix different issues.

With these changes, it is almost possible to set `allWarningsAsErrors = true`. Remaining warnings I didn't know how to fix:
```
w: src/main/kotlin/org/rust/ide/inspections/RsRedundantElseInspection.kt: (62, 24): 'patList: List<RsPat>?' is deprecated. Support `RsOrPat`
w: src/main/kotlin/org/rust/ide/inspections/checkMatch/CheckMatchUtils.kt: (46, 26): 'patList: List<RsPat>' is deprecated. Support `RsOrPat`
w: src/main/kotlin/org/rust/ide/inspections/checkMatch/RsMatchCheckInspection.kt: (53, 43): 'patList: List<RsPat>' is deprecated. Support `RsOrPat`
w: src/main/kotlin/org/rust/ide/inspections/checkMatch/RsMatchCheckInspection.kt: (62, 31): 'patList: List<RsPat>' is deprecated. Support `RsOrPat`
w: src/main/kotlin/org/rust/lang/core/cfg/CFGBuilder.kt: (175, 31): 'patList: List<RsPat>?' is deprecated. Support `RsOrPat`
w: src/main/kotlin/org/rust/lang/core/cfg/CFGBuilder.kt: (594, 17): 'patList: List<RsPat>' is deprecated. Support `RsOrPat`
w: src/main/kotlin/org/rust/lang/core/types/borrowck/ExprUseWalker.kt: (385, 31): 'patList: List<RsPat>?' is deprecated. Support `RsOrPat`
w: src/main/kotlin/org/rust/lang/core/types/borrowck/ExprUseWalker.kt: (447, 13): 'patList: List<RsPat>' is deprecated. Support `RsOrPat`
w: src/main/kotlin/org/rust/lang/core/types/borrowck/ExprUseWalker.kt: (452, 13): 'patList: List<RsPat>' is deprecated. Support `RsOrPat`
```

cc @Undin who introduced this (and fixed many instances) in d30be52fb5ca1ceddb907deb55ee8593948ccd89.

### CODE REVIEW

**Comment by tamird** [comment]
(fixed now to stay compatible with 201)

File: src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt [hunk 1]
```diff
@@ -48,66 +48,65 @@ import org.rust.lang.utils.evaluation.evaluate
...
+                override fun visitRangeExpr(o: RsRangeExpr) = checkRangeExpr(it, o)
+                override fun visitTraitType(o: RsTraitType) = checkTraitType(it, o)
+                override fun visitSelfParameter(o: RsSelfParameter) = checkParamAttrs(it, o)
+                override fun visitValueParameter(o: RsValueParameter) = checkParamAttrs(it, o)
+                override fun visitVariadic(o: RsVariadic) = checkParamAttrs(it, o)
+                override fun visitPatStruct(o: RsPatStruct) = checkRsPatStruct(it, o)
+                override fun visitPatTupleStruct(o: RsPatTupleStruct) = checkRsPatTupleStruct(it, o)
+                override fun visitStructLiteralField(o: RsStructLiteralField) = checkReferenceIsPublic(o, o, it)
+            }
+        }.let(element::accept)
```
[the remark this answers is not in the document]
Comment (tamird) [reply src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt:109]: Well, initially I had a local variable here, but I ran into some version of this bug: https://youtrack.jetbrains.com/issue/KT-29595

The only way I could make it work was using this let form.

File: src/main/kotlin/org/rust/lang/core/macros/ExpandedMacroStorage.kt [hunk 2]
```diff
@@ -499,8 +500,9 @@ class SourceFile(
         checkIsSmartMode(project)
 
         val isExpansionFile = MacroExpansionManager.isExpansionFile(file)
-        val isIndexedFile = file.isValid && (
-            FileBasedIndexScanRunnableCollector.getInstance(project).shouldCollect(file) || isExpansionFile)
+        val isIndexedFile = file.isValid && (isExpansionFile || ProjectFileIndex.getInstance(project).let {
```
[the remark this answers is not in the document]
Comment (tamird) [reply src/main/kotlin/org/rust/lang/core/macros/ExpandedMacroStorage.kt:503]: What's so bad about `let`? FWIW I'm not really sure the logic here is correct; I just copied what was being done in FileBasedIndexScanRunnableCollector.

[the remark this answers is not in the document]
Comment (Undin) [reply src/main/kotlin/org/rust/lang/core/macros/ExpandedMacroStorage.kt:503]: Usually `let` makes code less readable and more complex (including this case). Also, it makes debugging harder

File: src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Undin) [reply src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt:109]: Or just add suppress annotation as I mentioned above 😄 
This warning is not related to real problem

File: src/main/kotlin/org/rust/lang/core/macros/ExpandedMacroStorage.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (tamird) [reply src/main/kotlin/org/rust/lang/core/macros/ExpandedMacroStorage.kt]: OK, done.

File: src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (tamird) [reply src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt]: OK, done.

File: src/main/kotlin/org/rust/lang/core/macros/VfsInternals.kt [hunk 3]
```diff
@@ -41,8 +41,14 @@ object VfsInternals {
     fun getContentHashIfStored(file: VirtualFile): HashCode? =
         file.getContentHashIfStored()?.let { HashCode.fromByteArray(it) }
 
-    fun calculateContentHash(fileContent: ByteArray): HashCode =
-        HashCode.fromByteArray(DigestUtil.calculateContentHash(CONTENT_HASH_DIGEST, fileContent))
+    /** [com.intellij.openapi.vfs.newvfs.persistent.FSRecords.calculateHash] */
+    fun calculateContentHash(fileContent: ByteArray): HashCode {
```
[the remark this answers is not in the document]
Comment (vlad20012) [reply src/main/kotlin/org/rust/lang/core/macros/VfsInternals.kt:45]: Seems OK

**Pull request merged**